src/jshint.js

```diff
@@ -3592,6 +3604,8 @@ var JSHINT = (function() {
     if (state.tokens.next.value === "extends") {
       advance("extends");
       c.heritage = expression(10);
+    } else {
+      c.heritage = null;
```

Comment:
it doesn't catch this highly contrived example
```
// jshint esnext: true
class a extends function(){ return null;}() {
  constructor() {
    super();
  }
}
```
but I think that is fine. If we take that into account we can't really do any static analysis of whether super is required/allowed.